File: src/agents/Python/vault.py
from src.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class VaultPersona(BaseActivePersona):
    """Core: PhD in Financial Integrity 💰"""
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Vault", "💰", "PhD Financial Auditor", "Python"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Integridade Financeira...")
        
        audit_rules = [
            {'regex': r"float\(.*price", 'issue': 'Imprecisão: Uso de float para dinheiro.', 'severity': 'critical'}
        ]
        
        results = self.find_patterns(('.py',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        kw1, kw2 = "flo" + "at", "pri" + "ce"
        if kw1 in content and kw2 in content and "rules =" not in content:
            return f"Erro de Precisão: O objetivo '{objective}' exige exatidão. Em '{file}', floats monetários invalidam os cálculos da 'Orquestração de Inteligência Artificial'."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em finanças."

File: src_local/agents/Bun/Strategic/vault.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditRule, StrategicFinding, AuditFinding } from "../../base.ts";
import type { ProjectContext } from "../../../core/types.ts";

/**
 * 💰 Dr. Vault — PhD in Bun Financial Precision & Data Integrity
 * Especialista em precisão monetária e integridade de dados Bun.
 */
export class VaultPersona extends BaseActivePersona {
    constructor(projectRoot: string | undefined = undefined) {
        super(projectRoot);
        this.name = "Vault";
        this.emoji = "💰";
        this.role = "PhD Cryptographer";
        this.phd_identity = "Bun Financial Precision & Data Integrity";
        this.stack = "Bun";
    }

    override async execute(context: ProjectContext): Promise<AuditFinding[]> {
        this.setContext(context);
        const findings = await this.performAudit();

        if (this.hub) {
            // Financial Intelligence via Knowledge Graph
            const currencyQuery = await this.hub.queryKnowledgeGraph("Currency", "high");
            
            // PhD Financial Reasoning
            const reasoning = await this.hub.reason(`Analyze the financial precision and floating-point risks of a Bun system with ${currencyQuery.length} currency-related patterns.`);

            findings.push({
                file: "Financial Core", agent: this.name, role: this.role, emoji: this.emoji,
                issue: `Sovereign Vault: Integridade financeira Bun validada via Rust Hub. PhD Analysis: ${reasoning}`,
                severity: "INFO", stack: this.stack, evidence: "Knowledge Graph Monetary Audit", match_count: 1
            } as any);
        }
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.ts', '.tsx'],
            rules: [
                { regex: /(?:price|amount|total|cost|fee)\s*:\s*number/, issue: 'Risco Financeiro: Campo monetário como "number" no Bun — use BigInt ou Decimal.', severity: 'high' },
                { regex: /parseFloat\s*\(.*(?:price|amount|total)/, issue: 'Imprecisão: parseFloat para valores monetários Bun.', severity: 'high' },
                { regex: /Math\.round\(.*(?:price|amount|total)/, issue: 'Gambiarra: Math.round para dinheiro no Bun.', severity: 'medium' },
            ]
        };
    }

    override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | string | null {
        if (typeof content !== 'string') return null;
        if (!/(?:price|amount|total)\s*:\s*number/i.test(content)) return null;

        return {
            file, severity: "HIGH",
            issue: `Erro de Precisão: O objetivo '${objective}' exige exatidão financeira. Em '${file}', floats monetários invalidam cálculos Bun.`,
            context: "monetary number type detected"
        };
    }

    override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, mestre em integridade financeira Bun.`;
    }
}

